### PR TITLE
Stage slbeta4 sync

### PR DESCRIPTION
**Pull request by azinneera**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by MaryamZi — CHANGES_REQUESTED** [review]
Need to revert https://github.com/ballerina-platform/ballerina-lang/pull/33967/commits/9583fc5c4fac5e7bd30681547251f5ed910053e9 before we merge this PR.

**Comment by MaryamZi** [comment]
> Need to revert [9583fc5](https://github.com/ballerina-platform/ballerina-lang/commit/9583fc5c4fac5e7bd30681547251f5ed910053e9) before we merge this PR.

Done in https://github.com/ballerina-platform/ballerina-lang/pull/33975.

File: .github/workflows/publish_timestamped_release.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+      - stage-slbeta4
```
Comment (MaryamZi) [inline .github/workflows/publish_timestamped_release.yml:7]:
```suggestion
```

File: .github/workflows/pull_request_ubuntu_build.yml [hunk 2]
```diff
@@ -8,8 +8,7 @@ on:
       - release-stage
       - stage
       - ballerina-1.2.x
-      - stage-slbeta2
-      - 2.0.0-beta3-dev
+      - stage-slbeta4
```
Comment (MaryamZi) [inline .github/workflows/pull_request_ubuntu_build.yml:11]:
```suggestion
```

File: .github/workflows/pull_request_windows_build.yml [hunk 3]
```diff
@@ -8,8 +8,7 @@ on:
       - release-stage
       - stage
       - ballerina-[0-9]+.[0-9]+.x
-      - stage-slbeta2
-      - 2.0.0-beta3-dev
+      - stage-slbeta4
```
Comment (MaryamZi) [inline .github/workflows/pull_request_windows_build.yml:11]:
```suggestion
```

**Review by MaryamZi — APPROVED** [review]

**Review by MaryamZi — APPROVED** [review]
Front-end conflict resolving changes were reviewed via the individual PRs.

**Review by nadeeshaan — APPROVED** [review]

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by pcnfernando — APPROVED** [review]

**Comment by azinneera** [comment]
LGTM

**Review by lochana-chathura — APPROVED** [review]

**Review by Dilhasha — APPROVED** [review]

**Review by DamithaSenevirathne — APPROVED** [review]

**Pull request merged**